### PR TITLE
Erdikose/fix/auto updater

### PR DESCRIPTION
**Pull request by erdkse**
This PR...

## Changes

-

## Fixes

-

## How to test it

-

## Screenshots

-

## Checklist

- [ ] tested locally
- [ ] added new dependencies
- [ ] updated the docs
- [ ] added a test

### CODE REVIEW

File: .nvmrc [hunk 1]
```diff
@@ -1 +1 @@
-16.16.0
+18.7.0
```
Comment (erdkse) [inline .nvmrc:1]: Updated node version. Any reason to not update?

File: electron/app/autoUpdater.ts [hunk 2]
```diff
@@ -18,4 +22,11 @@ if (process.platform === 'win32') {
 }
 autoUpdater.logger = console;
 
+if (isDev) {
```
Comment (erdkse) [inline electron/app/autoUpdater.ts:25]: To test auto-updater in development mode

File: electron/app/openApplication.ts [hunk 3]
```diff
@@ -46,8 +46,8 @@ export const openApplication = async (givenPath?: string) => {
   });
 
   app.on('window-all-closed', () => {
-    if (process.platform !== 'darwin') {
-      app.quit();
+    if (!isDev) {
+      app.exit();
```
Comment (erdkse) [inline electron/app/openApplication.ts:50]: To exit the app on close

File: src/utils/logToFile.ts [hunk 4]
```diff
@@ -10,8 +10,4 @@ electronLog.transports.file.maxSize = 2 * 1024 * 1024;
 // one file in stand of main.log & renderer.log
 // electronLog.transports.file.fileName = 'logs.log';
 
-if (!process.env.NODE_ENV || process.env.NODE_ENV === 'development') {
```
Comment (erdkse) [inline src/utils/logToFile.ts:13]: Logs are not visible in the production environment too, Either set NODE_ENV or we should remove this block

File: dev-app-update.yml [hunk 5]
```diff
@@ -0,0 +1,3 @@
+owner: kubeshop
```
Comment (erdkse) [inline dev-app-update.yml:1]: To test auto-updater in development mode

File: .nvmrc (hunk 1 above)
Comment (erdkse) [inline .nvmrc:1]: Any reason to not update?

File: package.json [hunk 6]
```diff
@@ -196,7 +196,8 @@
     "release:minor": "standard-version --release-as minor",
     "release:major": "standard-version --release-as major",
     "ui-test": "xvfb-maybe npx playwright test --workers=1",
-    "custom-script": "npx ts-node src/scripts/index.ts"
+    "custom-script": "npx ts-node src/scripts/index.ts",
+    "sign-dev-electron-app": "codesign --deep --force --verbose --sign - node_modules/electron/dist/Electron.app"
```
Comment (erdkse) [inline package.json:200]: It allows us to sign application on development mode

File: src/utils/telemetry.ts [hunk 7]
```diff
@@ -29,6 +33,7 @@ export const trackError = (error: any) => {
     segmentClient?.track({
       event: 'Error',
       properties: error,
+      userId: machineId,
```
Comment (devcatalin) [inline src/utils/telemetry.ts:36]: good catch with this ones!

**Review by devcatalin — APPROVED** [review]

**Pull request merged**